src/button/index.tsx

```diff
@@ -52,26 +48,18 @@ const StyledLoading = styled("button")`
 
 export const PrimaryButton: FC<
   ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>
-> = ({ text, className = "", ...props }) => {
-  return (
-    <StyledPrimary className={className} {...props}>
-      {text}
-    </StyledPrimary>
-  );
-};
+> = ({ text, loading = false, className = "", ...props }) => {
+
+  if (!loading) {
+    return (
+      <StyledPrimary className={className} {...props}>
+        {text}
+      </StyledPrimary>
+    );
+  }
```

Comment:
use conditional  rendering instead of if statement and make sure to render the loading gif once if the loading is set to true

Reply:
Resolved the conflicts, added conditional rendering as well as that loading gif.